src/arch/Lights/LightsDriver_LinuxPacDrive.cpp

```diff
@@ -163,120 +172,152 @@ void LightsDriver_LinuxPacDrive::Set( const LightsState *ls )
+            return;
+        }
+
+        if (PACDRIVE_VENDOR_ID == desc.idVendor &&
+            PACDRIVE_PRODUCT_ID <= desc.idProduct &&
+            PACDRIVE_PRODUCT_ID + 8 > desc.idProduct)
+        {
+            Device = dev;
+            LOG->Info("PacDrive device was found vid: 0x%04x pid: 0x%04x", desc.idVendor, desc.idProduct);
+            return;
+        }
+    }
+
+    LOG->Warn("PacDrive was not found!");
+    Device = NULL;
+
+    libusb_free_device_list(devs, 1);
 }
 
 void LightsDriver_LinuxPacDrive::OpenDevice()
 {
-	CloseDevice();
-
-	if ( !Device ) return;
-
-	DeviceHandle = usb_open( Device );
-
-	if ( DeviceHandle == NULL ) {
-		LOG->Warn( "libusb: usb_open: %s", usb_strerror() );
-		return;
-	}
-
-	// The device may be claimed by a kernel driver. Attempt to reclaim it.
-	for ( unsigned iface = 0; iface < Device->config->bNumInterfaces; iface++ )
-	{
-		int result = usb_detach_kernel_driver_np( DeviceHandle, iface );
-
-		// device doesn't understand message, no attached driver, no error -- ignore these
-		if( result == -EINVAL || result == -ENODATA || result == 0 )
-			continue;
-
-		/* we have an error we can't handle; try and get more info. */
-		LOG->Warn( "usb_detach_kernel_driver_np: %s\n", usb_strerror() );
-
-		// on EPERM, a driver exists and we can't detach - report which one
-		if ( result == -EPERM )
-		{
-			char szDriverName[16];
-			strcpy( szDriverName, "(unknown)" );
-			usb_get_driver_np(DeviceHandle, iface, szDriverName, 16);
-
-			LOG->Warn( "(cannot detach kernel driver \"%s\")", szDriverName );
-		}
-
-		CloseDevice();
-		return;
-	}
-
-	if ( usb_set_configuration( DeviceHandle, Device->config->bConfigurationValue) ) {
-		LOG->Warn( "libusb: usb_set_configuration: %s", usb_strerror() );
-		CloseDevice();
-		return;
-	}
-
-	// attempt to claim all interfaces for this device
-	for ( unsigned i = 0; i < Device->config->bNumInterfaces; i++ )
-	{
-		if ( usb_claim_interface( DeviceHandle, i ) ) {
-			LOG->Warn( "Libusb: usb_claim_interface(%i): %s", i, usb_strerror() );
-			CloseDevice();
-			return;
-		}
-	}
+    CloseDevice();
+
+    if (!Device) return;
+
+    libusb_context* context = USBContext::getInstance().getContext();
+    if (!context)
+    {
+        LOG->Warn("libusb: Failed to initialize context");
+        return;
+    }
+
+    int result;
+    libusb_device **devs;
+
+    libusb_set_option(context, LIBUSB_OPTION_LOG_LEVEL, LIBUSB_LOG_LEVEL_INFO);
+    ssize_t cnt = libusb_get_device_list(context, &devs);
+    if (cnt < 0)
+    {
+        LOG->Warn("libusb_get_device_list: %s", libusb_error_name(cnt));
+        return;
+    }
+
+    // Check PID 1500 thru 1507
+    for (int pid = 0x1500; pid <= 0x1507; ++pid)
+    {
+        DeviceHandle = libusb_open_device_with_vid_pid(context, 0xD209, pid);
+        if (DeviceHandle == NULL)
+        {
+            LOG->Warn("libusb_open_device_with_vid_pid: Checked PID 1500 thru 1507, but did not find PacDrive");
```

Comment:
I don't know the history behind `PACDRIVE_PRODUCT_ID + 8`, but I can guess that PacDrive devices were made with different product version, so we scan for the first available device. I don't think it is a warning, if we couldn't find a device with pid 0x1500. For example, I have pid 0x1505.

Also, Trace? Info?
`LOG->Trace("libusb_open_device_with_vid_pid: Couldn't open device with PID %d", pid);`
or something like this